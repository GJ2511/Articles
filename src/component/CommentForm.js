import React, { useState } from 'react';
import PropTypes from 'prop-types';

function CommentForm({ handleCommentSubmit, requesting }) {
    const [text, setText] = useState('');

    return (
        <>
            <textarea
                className="form-control rounded-pill"
                value={text}
                placeholder="write a comment..."
                rows="3"
                disabled={requesting}
                onChange={(evt) => setText(evt.target.value)}
            ></textarea>
            <br />
            <button
                type="button"
                className="btn btn-info float-right rounded-pill"
                onClick={() => {
                    handleCommentSubmit(text);
                    setText('');
                }}
                disabled={requesting || text.length === 0}
            >
                Post
            </button>
        </>
    );
}

CommentForm.propTypes = {
    handleCommentSubmit: PropTypes.func.isRequired,
    requesting: PropTypes.bool.isRequired,
};

export default CommentForm;
