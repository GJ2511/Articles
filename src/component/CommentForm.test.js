import React from 'react';

import CommentForm from './CommentForm';

const props = {
    handleCommentSubmit: jest.fn(),
    requesting: false,
};

describe('<CommentForm tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<CommentForm {...props} />)).toMatchSnapshot();
    });
});
