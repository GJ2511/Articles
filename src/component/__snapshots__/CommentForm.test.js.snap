// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm tests /> match snapshot 1`] = `
<Fragment>
  <textarea
    className="form-control rounded-pill"
    disabled={false}
    onChange={[Function]}
    placeholder="write a comment..."
    rows="3"
    value=""
  />
  <br />
  <button
    className="btn btn-info float-right rounded-pill"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Post
  </button>
</Fragment>
`;
